토큰 재발급 테스트

10kseok opened the pull request:
토큰 재발급 테스트들을 위한 설정을 추가해주었습니다.

10kseok committed:
feat: 🎸수정 로고 이미지 동적으로 크기 변환

로그인 페이지에서 로고가 화면이 작아질 시에 짤리는 이슈가 있어
동적으로 변환되게 변경하였습니다.

10kseok committed:
feat: 🎸수정 로고 이미지 동적으로 크기 변환

로그인 페이지에서 로고가 화면이 작아질 시에 짤리는 이슈가 있어
동적으로 변환되게 변경하였습니다.

10kseok committed:
feat: 🎸 추가 예외처리 로깅

백엔드에서 사용하던 예외처리 로깅 로직을 프론트에도 적용시켰습니다.

10kseok committed:
feat: 🎸 추가 예외처리 로깅

백엔드에서 사용하던 예외처리 로깅 로직을 프론트에도 적용시켰습니다.

10kseok committed:
chore: 🤖 삭제 인증객체 패키지 생성 및 코드 정리

기존 model에 vo 패키지는 서버 통신시 사용되는 객체라서 인증에 쓰이는 인증 객체만을 따로 담아둘 패키지를 만들어 관리할  수 있게 하였습니다.

10kseok committed:
chore: 🤖 삭제 인증객체 패키지 생성 및 코드 정리

기존 model에 vo 패키지는 서버 통신시 사용되는 객체라서 인증에 쓰이는 인증 객체만을 따로 담아둘 패키지를 만들어 관리할  수 있게 하였습니다.

10kseok committed:
refactor: 💡 추가 직렬화 필수값과 RefreshToken 유틸형 메서드 작성

Custom Exception에서 누락돼있던 직렬화 버전 ID를 추가해주었고
편리하게 로그를 찍기위해 toString을 정의해주었습니다.

10kseok committed:
refactor: 💡 추가 직렬화 필수값과 RefreshToken 유틸형 메서드 작성

Custom Exception에서 누락돼있던 직렬화 버전 ID를 추가해주었고
편리하게 로그를 찍기위해 toString을 정의해주었습니다.

10kseok committed:
fix: 🐛 수정 로깅 PointCut 변경

백엔드에서 가져온 PointCut을 프론트 환경에 맞게 변경하였습니다.

10kseok committed:
fix: 🐛 수정 로깅 PointCut 변경

백엔드에서 가져온 PointCut을 프론트 환경에 맞게 변경하였습니다.

10kseok committed:
refactor: 💡 추가 로직 내부에 assert문 추가 및 예외 처리 강화

로그인 로직에서 에러들을 검출하기 위해 assert문을 추가했습니다.
로그인시 예외 발생 시점을 알아내기 위해 예외 처리를 상세하게 작성하였습니다.

10kseok committed:
refactor: 💡 추가 로직 내부에 assert문 추가 및 예외 처리 강화

로그인 로직에서 에러들을 검출하기 위해 assert문을 추가했습니다.
로그인시 예외 발생 시점을 알아내기 위해 예외 처리를 상세하게 작성하였습니다.

10kseok committed:
chore: 🤖 추가 운영환경에서 테스트를 위한 로그 작성

운영환경에서 테스트를 위해 세션 만료기간을 6분으로, 토큰 재발급 성공시에도 로그가 작성되도록 하였습니다.

10kseok committed:
chore: 🤖 추가 운영환경에서 테스트를 위한 로그 작성

운영환경에서 테스트를 위해 세션 만료기간을 6분으로, 토큰 재발급 성공시에도 로그가 작성되도록 하였습니다.

eora21 commented on src/main/java/store/cookshoong/www/cookshoongfrontend/auth/interceptor/TokenReissueInterceptor.java:
```diff
@@ -39,15 +45,36 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
+            AuthenticationResponseDto authResponse = tokenManagementService.reissueToken();
+            updateCurrentAccessToken(authResponse.getAccessToken());
+            tokenManagementService.saveRefreshToken(response, authResponse.getRefreshToken());
+            log.error("============ 토큰 재발급 성공 : {} ===============",
```
해당 사항은 error보단 debug, 또는 info로 해주시면 좋을 것 같아요

10kseok replied:
넵..! 토큰 재발급 테스트를 위해 error로 남겨놨어서 테스트 마친뒤 바로 제거하겠습니다!

10kseok replied:
런타임에 로깅레벨을 변경하도록 하는 기능을 추가하면서 해당 부분을 다시 조정해주었습니다!

eora21 commented on src/main/java/store/cookshoong/www/cookshoongfrontend/auth/interceptor/TokenReissueInterceptor.java:
```diff
@@ -69,5 +96,7 @@ public void updateCurrentAccessToken(String accessToken) {
         JwtAuthentication oldAuthentication = (JwtAuthentication) SecurityContextHolder.getContext()
             .getAuthentication();
         oldAuthentication.updateAccessToken(accessToken);
+        log.error("========== 재발급된 액세스 토큰 확인 : {} ================", SecurityContextHolder.getContext()
```
해당 사항은 error보단 debug, 또는 info로 해주시면 좋을 것 같아요

10kseok replied:
이것또한 토큰 재발급 테스트를 위해 error로 남겨놨어서 테스트 마친뒤 바로 제거하겠습니다!

10kseok replied:
런타임에 로깅레벨을 변경하도록 하는 기능을 추가하면서 해당 부분을 다시 조정해주었습니다!

10kseok committed:
feat: 🎸 추가 운영환경에서 로깅레벨 변경 기능 구현

운영환경에서 로깅레벨을 런타임에 바꿀 수 있도록 하는 기능을 구현했습니다.
이에 따라 로그레벨이 알맞지 않았던 부분들도 수정하였습니다.

10kseok committed:
feat: 🎸 추가 운영환경에서 로깅레벨 변경 기능 구현

운영환경에서 로깅레벨을 런타임에 바꿀 수 있도록 하는 기능을 구현했습니다.
이에 따라 로그레벨이 알맞지 않았던 부분들도 수정하였습니다.

10kseok committed:
Merge branch 'feature/adjust-dynamic-image-for-login' of https://github.com/nhnacademy-be3-CookShoong/cookshoong-frontend into feature/adjust-dynamic-image-for-login

10kseok committed:
refactor: 💡 수정 로그 레벨 조회 방식 변경

로그 레벨을 보여주는 로직에서 리스트 생성하는게 불필요하다고 생각하여 바로 String 값들을 합치는 로직으로 변경하였습니다.

10kseok committed:
refactor: 💡 수정 로그 메세지 변경

RestTemplate을 이용하는 코드 내부에 로그들의 메세지들을 변경해주었습니다.
Gateway에서 발생한 문제인지 인증서버 또는 백엔드 서버에서 발생한 문제인지를 모르기에 외부 서버에서 발생한 문제로 로그 메세지를 변경하였습니다.
토큰 재발급 중 로그아웃되는 상황에서 실제 로그아웃과 구분하여 알기위해 로그를 추가했습니다.

seong1542 reviewed (approved):
확인했습니다 오늘 테스트 해봅시다!
